### PR TITLE
Unify prepare_checkpoint.py

### PR DESCRIPTION
**Pull request by daniil-lyakhov**
* Add param `--model-type` to make it possible to choose between object detection and segmentation models
* Add sanity test (running time~12 sec on local machine) which check compressed model creates without errors

### CODE REVIEW

**Comment by daniil-lyakhov** [comment]
Should be merged after #775 to prevent merge conflicts

File: tests/tensorflow/test_sanity_sample.py [hunk 1]
```diff
@@ -362,3 +364,28 @@ def test_export_with_resume(_config, tmp_path, export_format, _case_common_dirs)
...
+def test_prepare_checkpoint(model_type, config_path):
+    dirpath = tempfile.mkdtemp()
+    args = {
+        '--model-type': model_type,
+        '--config': config_path,
+        '--checkpoint-save-dir': dirpath,
+        '--resume': dirpath,
+    }
+
+    prepare_checkpoint_main(convert_to_argv(args))
```
Comment (negvet) [inline tests/tensorflow/test_sanity_sample.py:390]: Maybe it worth adding a test that loads the saved model, which makes sure that the model is saved in a proper way. What do you think?

Comment (alexsu52) [reply tests/tensorflow/test_sanity_sample.py:390]: @daniil-lyakhov  please, estimate the impact on pre-commit time.

Comment (daniil-lyakhov) [reply tests/tensorflow/test_sanity_sample.py:390]: Ok, I'll add test and check running time

Comment (daniil-lyakhov) [reply tests/tensorflow/test_sanity_sample.py:408]: I've added evaluation for prepare checkpoint. test_prepare_checkpoint took ~48 second to execute it locally

**Comment by daniil-lyakhov** [comment]
Precommit time
|develop   | current PR | diff  |
| ~37 min | ~38min     | ~1 min |

**Review by alexsu52 — APPROVED** [review]

**Pull request merged**